src/features/island/fisherman/FishingChallenge.tsx

```diff
@@ -0,0 +1,339 @@
+  let randomAngle;
+
+  do {
+    randomAngle = Math.random() * 270; // Generate a random angle between 0 and 270
+  } while (Math.abs(randomAngle - existingAngle) < minDistance);
+
+  return randomAngle;
+}
+
+const extractAngle = (transformString: string) => {
+  const match = transformString.match(/[-+]?\d*\.\d+|\d+/);
+  if (match && match[0]) {
+    return parseFloat(match[0]);
+  }
+  return 0; // Default value if the match fails
+};
+
+// Converts 354 -> 6
+const normaliseAngle = (angle: number) => {
+  return ((angle % 360) + 360) % 360;
+};
+
+type Attempt = "hit" | "miss";
+
+function getSpeed(attempts: number) {
+  if (attempts > 20) {
+    return 1000;
+  }
+
+  if (attempts > 15) {
+    return 1200;
+  }
+
+  if (attempts > 10) {
+    return 1500;
+  }
+
+  if (attempts > 7) {
+    return 2000;
+  }
+
+  if (attempts > 3) {
+    return 2500;
+  }
+
+  return 3000;
+}
+
+// Difficult to health
+const FISH_HEALTH: Record<number, number> = {
+  1: 10,
+  2: 13,
+  3: 16,
+  4: 20,
+  5: 25,
+};
+
+interface Props {
+  onCatch: () => void;
+  onMiss: () => void;
+  difficulty?: number;
+  fishName: FishName | MarineMarvelName;
+}
+
+export const FishingChallengeIntro: React.FC<{ onNext: () => void }> = ({
+  onNext,
+}) => {
+  return (
+    <>
+      <div className="p-2">
+        <p className="text-sm">A powerful catch awaits!</p>
+        <p className="text-sm">Use all your strength to reel it in.</p>
+        <p className="text-sm">Stop the green bar on the fish to succeed</p>
+        <p className="text-sm">Be quick - 3 wrong tries, and it escapes!</p>
+      </div>
+      <Button onClick={onNext}>Next</Button>
+    </>
+  );
+};
+
+export const FishingChallenge: React.FC<Props> = ({
+  onCatch,
+  onMiss,
+  difficulty = 1,
+  fishName,
+}) => {
+  const [showIntro, setShowIntro] = useState(true);
+  const [state, setState] = useState<"idle" | "playing">("idle");
+
+  const [tentacleAngle, setTentacleAngle] = useState(getRandomAngle(150, 0));
+  const [attempts, setAttempts] = useState<Attempt[]>([]);
+
+  const direction = useRef<"clockwise" | "anticlockwise">("clockwise");
+  const [castFrom, setCastFrom] = useState(0);
+  const [castTo, setCastTo] = useState(360);
+
+  const health = FISH_HEALTH[difficulty] ?? 10;
+
+  const changeDirection = (fromDegrees: number) => {
+    const to =
+      direction.current === "clockwise" ? fromDegrees - 360 : fromDegrees + 360;
+    setCastTo(to);
+    setCastFrom(fromDegrees);
+    direction.current =
+      direction.current === "anticlockwise" ? "clockwise" : "anticlockwise";
+  };
+
+  const greenBarProps = useSpring({
+    from: { transform: `rotate(${castFrom}deg)` },
+    to: { transform: `rotate(${castTo}deg)` },
+    onRest: () => {
+      if (misses >= 2) {
+        onMiss();
+      }
+
+      changeDirection(castTo);
+      setAttempts((prev) => [...prev, "miss"]);
+    },
+    config: {
+      duration: getSpeed(attempts.length),
+    },
```

Comment:
```suggestion
    config: {
      duration: 5000,
      mass:0,
      tension:0,
      friction:0
      
    },
```

Consider un-setting the default spring configuration on the green bar